### PR TITLE
feat: show Heroic and Mythic dungeon runs

### PR DESCRIPTION
**Pull request by ntowle**

### CODE REVIEW

File: SavedInstances/Modules/MythicPlus.lua [hunk 1]
```diff
@@ -187,12 +200,32 @@ do
...
     end
+
+    -- add Mythic 0 and Heroic runs
+    local numHeroic, numMythic = C_WeeklyRewards_GetNumCompletedDungeonRuns()
+    local _, _, rewardSeasonID = C_MythicPlus_GetCurrentSeasonValues()
+    for i = 1, numMythic do
+      runHistory[#runHistory + 1] = {
+        name = WEEKLY_REWARDS_MYTHIC:format(WeeklyRewardsUtil_MythicLevel),
+        rewardLevel = ItemLevelsBySeason[rewardSeasonID] and ItemLevelsBySeason[rewardSeasonID].MYTHIC or 0,
+        level = L["M"],
```
Comment (LiangYuxuan) [inline SavedInstances/Modules/MythicPlus.lua:218]: Difficulty abbr. won't allow locale, use normal string.

File: SavedInstances/Modules/MythicPlus.lua [hunk 2]
```diff
@@ -187,12 +200,32 @@ do
...
+      }
+    end
+    for i = 1, numHeroic do
+      runHistory[#runHistory + 1] = {
+        name = WEEKLY_REWARDS_HEROIC,
+        rewardLevel = ItemLevelsBySeason[rewardSeasonID] and ItemLevelsBySeason[rewardSeasonID].HEROIC or 0,
+        level = L["H"],
```
Comment (LiangYuxuan) [inline SavedInstances/Modules/MythicPlus.lua:225]: Difficulty abbr. won't allow locale, use normal string.

**Review by LiangYuxuan — CHANGES_REQUESTED** [review]

**Comment by LiangYuxuan** [comment]
Thank you! For the heroic and mythic 0 item level part, did you try `C_MythicPlus.GetRewardLevelForDifficultyLevel`? I'm no quite sure but I don't want to have one more thing to update every season.

**Comment by ntowle** [comment]
> Thank you! For the heroic and mythic 0 item level part, did you try `C_MythicPlus.GetRewardLevelForDifficultyLevel`? I'm no quite sure but I don't want to have one more thing to update every season.

I did try, it doesn't return anything for values below 2. Maybe in the future it will.

There is `C_WeeklyRewards.GetNextActivitiesIncrease(2, 0)` which is "how do I increase my Heroic reward" and that returns the 450 item level from Mythic, but there's no way I can see to get the 441 item level value, so figured it's better to just include both.

**Comment by LiangYuxuan** [comment]
Maybe `C_WeeklyRewards.GetNextActivitiesIncrease(2, -1)`?

**Comment by ntowle** [comment]
> Maybe `C_WeeklyRewards.GetNextActivitiesIncrease(2, -1)`?

[21:04:40] Dump: value=C_WeeklyRewards.GetNextActivitiesIncrease(2, -1)
[21:04:40] [1]=false

**Comment by LiangYuxuan** [comment]
GG, BLZ just don't want we to get it.

**Review by LiangYuxuan — APPROVED** [review]

**Pull request merged**